augur/utils.py

```diff
@@ -126,7 +128,17 @@ def write_json(data, file, indent=(None if os.environ.get("AUGUR_MINIFY_JSON") e
         data["generated_by"] = {"program": "augur", "version": get_augur_version()}
     with open_file(file, 'w', encoding='utf-8') as handle:
         sort_keys = False if isinstance(data, OrderedDict) else True
-        json.dump(data, handle, indent=indent, sort_keys=sort_keys, cls=AugurJSONEncoder)
+        try:
+            json.dump(data, handle, indent=indent, sort_keys=sort_keys, cls=AugurJSONEncoder)
+        except TypeError as e:
+            if "'<' not supported" in str(e):
+                handle.seek(0)
+                handle.truncate(0)
+                print("WARNING: When writing JSON, could not sort JSON keys. A tree node is likely missing a name. Retrying without sorting keys.", file=sys.stderr)
```

Comment:
`write_json` is not specific to tree node JSONs and the `TypeError` can also be due to other types (e.g. `str` vs `int`), so this hint might be misleading/confusing. 

How about just reporting the underlying error:

```suggestion
                print(f"WARNING: When writing JSON, could not sort JSON keys due to the error: {e}. Retrying without sorting keys.", file=sys.stderr)
```